### DISCUSSION TITLE
Few years of transactions. GUI is getting slow.

### DISCUSSION THREAD

**Discussion by thehijacker** [firefly-iii/firefly-iii#9554] [Firefly III]
Using Waterfly III as client. Opening Transaction tab takes around 10 seconds to open. Same slow actions I can see in the Web GUI when I open account transactions. Everything is at latest version. How to start debugging what call is taking so long to process and what can I do to speed it up?

Thank you.

**Comment by JC5** [comment]
Hey, thanks for opening an issue. Specific pages are slow, yes. The queries are less than optimal :(

I hope to improve this soon, but for now, it's what it is.

**Reply by thehijacker** [reply]
Would it help adding additional indexes to the MariaDB database I am using at backend?

**Reply by JC5** [reply]
Most tables already have the correct indices, but let me know if you feel there are missing ones.
